lib/registration.pm

```diff
@@ -150,6 +150,10 @@ sub add_suseconnect_product {
     $version //= '${VERSION_ID}';
     $arch    //= '${CPU}';
     $params  //= '';
+    if ($name =~ /PackageHub/ && is_sle '>=15') {
+        return 0 unless (script_run('SUSEConnect -p PackageHub/${VERSION_ID}/${CPU}', 300) != 0);
```

Comment:
This seems too complicated to understand while no offering any benefit over a proposal using best practices (here return is used like the old times of **goto**), why not using a proper if? It is better to be explicit than implicit.

```perl
my $result = script_run("SUSEConnect -p $name/$version/$arch $params", $timeout);

if ($result != 0 && is_sle('>=15') && $name =~ /PackageHub/) {
    record_soft_failure 'bsc#1124318 - Fail to get PackageHub Pool Metadata - running the command again as a workaround';
    assert_script_run("SUSEConnect -p $name/$version/$arch $params", $timeout);
}
```

Reply:
actually about the use of the parameters `$name/$version/$arch` you are right. Updated the PR.

However I don't understand what you mean with "while no offering any benefit over a proposal using best practices".

For the early-return I don't think that we should wrap the complete rest of the function body in an if. Yes, it's better to be explicit, this is what I tried to do with an explicit early return :)

Reply:
You already wrap the workaround in a if. And a `return + unless + !=` in the same line is just like making complicated something that isn't.

Which one of these would help to understand better the reason and problem behind?

The explicit, which represents retrying after a failure, if it fails
```perl
my $result = script_run("SUSEConnect -p $name/$version/$arch $params", $timeout);

if ($result != 0 && is_sle('>=15') && $name =~ /PackageHub/) {
    record_soft_failure 'bsc#1124318 - Fail to get PackageHub Pool Metadata - running the command again as a workaround';
    assert_script_run("SUSEConnect -p $name/$version/$arch $params", $timeout);
}
```

Or the one-liner, which assumes the failure always, apply the workaround first and then the normal behavior, changing the order of cause-effect so that a tweak (return) is needed.
```perl
if ($name =~ /PackageHub/ && is_sle '>=15') {
    return 0 unless (script_run('SUSEConnect -p PackageHub/$version/$arch', 300) != 0);
    record_soft_failure 'bsc#1124318 - Fail to get PackageHub Pool Metadata - running the command again as a workaround';
}

assert_script_run("SUSEConnect -p $name/$version/$arch $params", $timeout);
```

Think about all possible people that would need to read or maintain the code.

Reply:
ok, updated to avoid the return+unless within the if

Reply:
@SergioAtSUSE I hope you like it as being more explicit now also with even less duplication using the `$cmd` variable